File: Problems/08-Search-Insert-Position/Search-Insert-Position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int left = 0;
        int right = nums.size()-1;
        while (left <= right)
        {
            int mid = (right - left) / 2 + left;
            if (nums[mid] == target) return mid;
            if (nums[mid] < target) left = mid + 1;
            else right = mid - 1;
        }

        if (right >= 0 && nums[right] > target) return right;
        else return right + 1;
    }
};

File: Problems/08-Search-Insert-Position/Search-Insert-Position.ts
function searchInsert(nums: number[], target: number): number {

    let left: number = 0;
    let right: number = nums.length;
    let mid: number
    while (left < right) {
        mid = left + Math.floor((right - left) / 2);
        if (nums[mid] >= target) right = mid;
        else left = mid + 1;

    }
    return left;

};